src/v2/Apps/Order/Routes/Shipping/index.tsx

```diff
@@ -146,211 +141,180 @@ export class ShippingRoute extends Component<ShippingProps, ShippingState> {
     }
   }
 
-  handleAddressDelete = (deletedAddressID: string) => {
-    const addressList = this.getAddressList()
+  const handleAddressDelete = (deletedAddressID: string) => {
+    const addressList = getAddressList()
 
     if (!addressList || addressList.length === 0) {
-      this.setState({
-        selectedAddressID: NEW_ADDRESS,
-        shippingQuotes: null,
-        shippingQuoteId: undefined,
-      })
-    } else if (this.state.selectedAddressID == deletedAddressID) {
-      this.selectSavedAddress(
+      setSelectedAddressID(NEW_ADDRESS)
+      setShippingQuotes(null)
+      setShippingQuoteId(undefined)
+    } else if (selectedAddressID == deletedAddressID) {
+      selectSavedAddress(
         addressList.find(address => address?.node?.isDefault)?.node?.internalID!
       )
     }
   }
 
-  componentDidMount() {
-    if (
-      this.isArtaShipping() &&
-      !this.isCreateNewAddress() &&
-      !this.state.shippingQuoteId
-    ) {
-      this.selectShipping()
-    }
-  }
-
-  getAddressList = () => this.props.me.addressConnection?.edges
+  const getAddressList = () => props.me.addressConnection?.edges
 
-  getOrderArtwork = () => this.props.order.lineItems?.edges?.[0]?.node?.artwork
+  const getOrderArtwork = () => props.order.lineItems?.edges?.[0]?.node?.artwork
 
-  isCreateNewAddress = () => this.state.selectedAddressID === NEW_ADDRESS
+  const isCreateNewAddress = () => selectedAddressID === NEW_ADDRESS
 
-  isArtaShipping = () => {
-    const addresses = this.getAddressList()
-    const artaShippingEnabled = !!this.getOrderArtwork()?.artaShippingEnabled
+  const isArtaShipping = () => {
+    const addresses = getAddressList()
+    const artaShippingEnabled = !!getOrderArtwork()?.artaShippingEnabled
 
-    const shippingCountry = this.isCreateNewAddress()
-      ? this.state.address.country
+    const shippingCountry = isCreateNewAddress()
+      ? address.country
       : addresses &&
         addresses.find(
-          address => address?.node?.internalID == this.state.selectedAddressID
+          address => address?.node?.internalID == selectedAddressID
         )?.node?.country
 
     return (
-      this.state.shippingOption === "SHIP" &&
+      shippingOption === "SHIP" &&
       artaShippingEnabled &&
       shippingCountry === "US"
     )
   }
 
-  onContinueButtonPressed = async () => {
-    if (this.isArtaShipping() && !!this.state.shippingQuoteId) {
-      this.selectShippingQuote()
+  useEffect(() => {
+    if (
+      isArtaShipping() &&
+      selectedAddressID &&
+      !shippingQuotes &&
+      !shippingQuoteId
+    ) {
+      selectShipping()
+    }
+  }, [shippingOption, shippingQuotes, shippingQuoteId])
```

Comment:
the hook here is missing one dependency, `isArtaShipping`.

if the function signature changes very often, you can wrap this function in a useCallback